tests/test_tester_utils.py

```diff
@@ -76,3 +81,22 @@ def test_get_last_group(self):
         # Test the case of no match
         last_group = get_last_group(bucket, "TestCam", "20110101")
         self.assertEqual(last_group, int(20110101) * 100_000)
+
+    def test_exposure_id_hsc(self):
+        group = "2023011100026"
+        # Need a Butler registry to test ExposureIdInfo
+        with tempfile.TemporaryDirectory() as repo:
+            butler = butler_tests.makeTestRepo(repo)
+            HyperSuprimeCam().register(butler.registry)
+            for record in butler.registry.queryDimensionRecords("instrument", dataId={"instrument": "HSC"}):
+                exp_max = record.exposure_max
+
+            _, str_exp_id, exp_id = make_exposure_id("HSC", int(group), 0)
+            butler_tests.addDataIdValue(butler, "visit", exp_id)
+            data_id = butler.registry.expandDataId({"instrument": "HSC", "visit": exp_id, "detector": 111})
+
+        self.assertEqual(str_exp_id, "HSCE%08d" % exp_id)
+        self.assertEqual(len(str_exp_id[4:]), 8)
```

Comment:
This is kind of redundant with the above.

Reply:
Actually, it's not: if `exp_id` is 9+ digits, then the formatting (here or in the application code) will dutifully create a string with 9+ digits, even though it wouldn't get parsed by `astro_metadata_translator`. I can add a comment to that effect.